File: BookStore.API/ClientApp/src/components/Books.js
import React, { Component } from 'react'
import { readCookie } from './CookieHandler'
import './styleBookCards.css'
import { ModalExample } from './Modal'

export class Books extends Component {
  static displayName = Books.name

  constructor (props) {
    super(props)
    this.state = { books: [], loading: true, collapsed: false }
    this.toggleModal = this.toggleModal.bind(this)
  }

  toggleModal () {
    const currentState = this.state.collapsed
    this.setState({ collapsed: !currentState })
  }

  componentDidMount () {
    this.populateBooksData()
  }

  static renderBooksTable (books) {
    return (
      <div className="BookCardsContainerGrid">
        <>
          {books.map((books) => (
            <div
              key={books.isbn}
              className="BookCard"
              style={{ minWidth: '95%', height: '600px' }}
            >
              <img
                className="card-img-top BookCardImg"
                src={books.imgLink}
                alt={books.imgLink}
              />
              <hr className="hr" />
              <div className="BookCardBody">
                <h4 className="card-title BookCardTitle NormalText" id="title">
                  {books.title}
                </h4>
                <ModalExample
                  imgLink={books.imgLink}
                  authorName={books.authorName}
                  desc={books.description}
                  genre={books.genreName}
                  isbn={books.isbn}
                  lang={books.languageName}
                  pages={books.pageNumber}
                  price={books.price}
                  publisher={books.publisherName}
                  publishingYear={books.publisherYear}
                  title={books.title}
                />
              </div>
            </div>
          ))}
        </>
      </div>
    )
  }

  render () {
    const contents = this.state.loading
      ? (
      <div className="d-inline-flex align-items-center">
        <div
          className="spinner-border ml-auto "
          role="status"
          aria-hidden="true"
        ></div>
      </div>
        )
      : (
          Books.renderBooksTable(this.state.books)
        )

    return (
      <div className="BaseSize">
        <h1 id="tabelLabel">All the books</h1>
        {contents}
      </div>
    )
  }

  async populateBooksData () {
    const splitUrl = window.location.href.split('/')
    let url

    if (splitUrl.length === 5) {
      url = `api/books/${splitUrl[4]}`
    } else {
      url = 'api/books'
    }

    try {
      const response = await fetch(url, {
        headers: {
          Authorization: `Bearer ${readCookie('token')}`
        }
      })
      const data = await response.json()
      this.setState({ books: data, loading: false })
    } catch (e) {}
  }
}

File: BookStore.APIDesktop/BookStore.APIDesktop/Database/Books.cs
﻿#nullable disable

namespace BookStore.APIDesktop
{
    public partial class Books
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public int AuthId { get; set; }
        public int GenreId { get; set; }
        public int Pagenumber { get; set; }
        public int LangId { get; set; }
        public string Isbn { get; set; }
        public string Description { get; set; }
        public string ImgLink { get; set; }
        public int PublisherId { get; set; }
        public int Price { get; set; }
        public int PublishingYear { get; set; }


        public string GenreName { get; set; }
        public string AuthorName { get; set; }
        public string LanguageName { get; set; }
        public string PublisherName { get; set; }
        public virtual Author Auth { get; set; }
        public virtual Genre Genre { get; set; }
        public virtual Language Lang { get; set; }
        public virtual Publisher Publisher { get; set; }
    }
}